consumer-services/src/main/scala/za/co/absa/spline/consumer/service/repo/ExecutionPlanRepositoryImpl.scala

```diff
@@ -134,6 +134,41 @@ class ExecutionPlanRepositoryImpl @Autowired()(db: ArangoDatabaseAsync) extends
           edges = edges)
     }
   }
+
+  override def getDataSources(value: String, access: Option[String])(implicit ec: ExecutionContext): Future[Array[String]] ={
+    val readResult = db.queryStream[String](
+      """
+        FOR d in dataSource
+        |FOR dp in depends
+        |FILTER dp._from == @planId AND dp._to == d._id
+        |RETURN d.uri
+        |""".stripMargin
+        .stripMargin,
+      Map("planId" -> "executionPlan".concat("/").concat(value)
+      )
+    )
+
+    val writeResult = db.queryStream[String](
+      """
+        FOR d in dataSource
+        |FOR aff in affects
+        |FILTER aff._from == @planId AND aff._to == d._id
+        |RETURN d.uri
+        |""".stripMargin
+        .stripMargin,
+      Map("planId" -> "executionPlan".concat("/").concat(value)
+      )
+    )
+
+    val totalResult = readResult.zip(writeResult).map{case(listOne,listTwo) => listOne ++ listTwo}
+
+    if(access.isEmpty)
+      totalResult.map(_.toArray)
+    else if(access.equals("read"))
```

Comment:
Consider using enums for this

Reply:
added , I added a separate enumerator class inside persistance/model . Let me know if I need to add it somewhere else. Thanks